### DISCUSSION TITLE
Problem formulation

### DISCUSSION THREAD

**Discussion by mssanamateen** [mssanamateen/Artificial-Intelligence-Lab#6] [Q&A] [answered]
What are the 4 components of a problem?

**Comment by CodeKatanaa** [comment, marked as answer]
In AI, a problem refers to a situation where an agent is tasked with finding a solution by moving from an initial state to a goal state using a set of actions and rules. The agent’s objective is to perform actions that will lead to the goal while considering the transitions resulting from each action.

A problem can be defined formally by five components:

**Initial State**: This is where the agent starts. It’s like the starting point of the problem, where the agent knows its current position or condition before taking any action.

**Actions**: These are the possible moves the agent can make. At each point, the agent has choices, and these actions determine how it can move from one state to another.

**Transition Model**: This explains how each action changes the current situation. When the agent takes an action, the transition model helps figure out what the new situation will be.

**Goal Test**: This checks whether the agent has achieved the goal. It’s like a success check, telling the agent if it has solved the problem and reached the final goal.

**Path Cost**: This value shows how expensive each action is. The agent uses this to find the most efficient way to reach the goal, usually by minimizing the total cost of actions.

**NOTE: This information is extracted from the book Artificial Intelligence: A Modern Approach by Stuart Russell and Peter Norvig**
